### DISCUSSION TITLE
High GPU usage. GPU on 100% most of time.

### DISCUSSION THREAD

**Discussion by luqsthunder** [hyprwm/Hyprland#4747] [Q&A] [answered]
Hi, guys. After I upgraded from Hyprland 0.34 to 0.35 I'm experiencing a lot of lag in apps such as Neovide, Vivaldi and Firefox. I've downgraded back to 0.34 since then, and the issue persisted. I've rolled back all the way to 0.29 and finally got rid of the lag (this [comment](https://github.com/hyprwm/Hyprland/issues/4670#issuecomment-1937838088) gave me the lead).

However, to use Neovide I need to launch it using render offload to Nvidia, otherwise the lag persists. The GPU usage goes up to 100% with a simple mouse movement in v0.35. And Neovide using the render offload on v0.35 stills with lags.

To render offload, I am using the flags: `__NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME <command>`

I tried to use the vars below after reading other discussions: ([4240](https://github.com/hyprwm/Hyprland/discussions/4240), [4575](https://github.com/hyprwm/Hyprland/discussions/4575))

```
debug { damage_tracking = 0} 
opengl { nvidia_anti_flicker = false }
misc { mouse_move_enables_dpms = true }
```
They do not solve completely, but it helps the lag not become too visible.

I am using Hyprland on hybrid graphics, iGPU intel TigerLake-H (i7-11800H) and NVIDIA RTX 3070. And before the date 2024/02/06 the system was really ok with the GPU usage between 70% ~ 80%.

The variables and envs from my config are below:

<details>

```
env = XDG_CURRENT_DESKTOP,Hyprland
env = XDG_SESSION_TYPE,wayland
env = XDG_SESSION_DESKTOP,Hyprland

env = LIBVA_DRIVER_NAME,nvidia
env = GBM_BACKEND,nvidia-drm
env = __GLX_VENDOR_LIBRARY_NAME,nvidia
env = WLR_NO_HARDWARE_CURSORS,1

env = __VK_LAYER_NV_optimus,NVIDIA_only
env = NVD_BACKEND,direct

env = GDK_BACKEND,wayland
env = QT_QPA_PLATFORM,wayland
env = QT_QPA_PLATFORMTHEME,qt5ct 
env = QT_WAYLAND_DISABLE_WINDOWDECORATION,1


env = MANGOHUD,1
env = MANGOHUD_DLSYM,1

opengl {
   nvidia_anti_flicker = false 
}

debug { 
   damage_tracking = 0 
}

input {
    kb_layout = us
    kb_variant = intl
    kb_model =
    kb_options =
    kb_rules =
    follow_mouse = 1

    touchpad {
        natural_scroll = yes 
    }

    sensitivity = 0 # -1.0 - 1.0, 0 means no modification.
    force_no_accel = 1
}

animations {
    # Some default animations, see https://wiki.hyprland.org/Configuring/Animations/ for more
    enabled = yes
    bezier = wind, 0.05, 0.9, 0.1, 1.05
    bezier = winIn, 0.1, 1.1, 0.1, 1.1
    bezier = winOut, 0.3, -0.3, 0, 1
    bezier = liner, 1, 1, 1, 1
    animation = windows, 1, 6, wind, slide
    animation = windowsIn, 1, 6, winIn, slide
    animation = windowsOut, 1, 5, winOut, slide
    animation = windowsMove, 1, 5, wind, slide
    animation = border, 1, 1, liner
    animation = borderangle, 1, 30, liner, loop
    animation = fade, 1, 10, default
    animation = workspaces, 1, 5, wind
}

dwindle {
    pseudotile = yes # master switch for pseudotiling. Enabling is bound to mainMod + P in the keybinds section below
    preserve_split = yes # you probably want this
}

master {
    new_is_master = true
}

gestures {
    workspace_swipe = off
}

misc {
    vrr = 0
    # vfr = false 
    # mouse_move_enables_dpms = true
}
```
</detais>

**Comment by vaxerski** [comment, marked as answer]
#4670 ?

**Reply by luqsthunder** [reply]
Thanks, @vaxerski, it solves. But the GPU continues to spikes or maintains on 100%, but without any lags. 

Here follows a graph, from nvtop, of my iGPU.
![image](https://github.com/hyprwm/Hyprland/assets/1775527/1b280505-4b56-40c3-bfaf-b6c1b90f8727)

Is it now a normal behavior, or am I missing something?
Maybe force the youtube decode to run on nvidia gpu?

**Reply by vaxerski** [reply]
if you're watching video with hw accel on a shit igpu it makes sense. Can't help much, I dont use such setups.

**Reply by luqsthunder** [reply]
Yeah, intel doesn’t have any good iGPU as far I remember. I've switched to the Hyprland on master branch (using now hyprland-git from Arch Linux AUR), looks more stable, but still getting GPU to 100%. I will try to figure out if I am missing something. Thanks for the quick reply.
